verticapy/learn/tools.py

```diff
@@ -0,0 +1,611 @@
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# |_     |~) _  _| _  /~\    _ |.
+# |_)\/  |_)(_|(_||   \_/|_|(_|||
+#    /
+#              ____________       ______
+#             / __        `\     /     /
+#            |  \/         /    /     /
+#            |______      /    /     /
+#                   |____/    /     /
+#          _____________     /     /
+#          \           /    /     /
+#           \         /    /     /
+#            \_______/    /     /
+#             ______     /     /
+#             \    /    /     /
+#              \  /    /     /
+#               \/    /     /
+#                    /     /
+#                   /     /
+#                   \    /
+#                    \  /
+#                     \/
+#                    _
+# \  / _  __|_. _ _ |_)
+#  \/ (/_|  | |(_(_|| \/
+#                     /
+# VerticaPy is a Python library with scikit-like functionality to use to conduct
+# data science projects on data stored in Vertica, taking advantage Vertica’s
+# speed and built-in analytics and machine learning features. It supports the
+# entire data science life cycle, uses a ‘pipeline’ mechanism to sequentialize
+# data transformation operations, and offers beautiful graphical options.
+#
+# VerticaPy aims to solve all of these problems. The idea is simple: instead
+# of moving data around for processing, VerticaPy brings the logic to the data.
+#
+#
+# Modules
+#
+# VerticaPy Modules
+from verticapy.toolbox import *
+
+#
+# ---#
+def check_model(name: str, cursor=None):
+    """
+---------------------------------------------------------------------------
+Checks if the model already exists.
+
+Parameters
+----------
+name: str
+    Model name.
+cursor: DBcursor, optional
+    Vertica DB cursor.
+    """
+    check_types([("name", name, [str],)])
+    cursor, conn = check_cursor(cursor)[0:2]
+    cursor.execute("SELECT * FROM MODELS WHERE model_name='{}'".format(name))
+    result = cursor.fetchone()
+    if result:
+        if conn:
+            conn.close()
+        raise NameError("The model '{}' already exists !".format(name))
+    else:
+        try:
+            cursor.execute(
+                "SELECT * FROM verticapy.models WHERE LOWER(model_name) = '{}'".format(
+                    str_column(name)
+                )
+            )
+            result = cursor.fetchone()
+        except:
+            result = []
+        if result:
+            raise NameError("The model '{}' already exists !".format(name))
```

Comment:
The `conn` will be left open if the program raises here. Yon can repeat lines 73 and 74 here.

Reply:
It is done and it will be pushed after the v0.5.0 release.

Reply:
Done;